notebooks/Connected Components -- FastSV.ipynb

```diff
@@ -0,0 +1,2433 @@
+   "metadata": {
+    "execution": {
+     "iopub.execute_input": "2021-11-09T18:51:47.675511Z",
+     "iopub.status.busy": "2021-11-09T18:51:47.675511Z",
+     "iopub.status.idle": "2021-11-09T18:51:47.688507Z",
+     "shell.execute_reply": "2021-11-09T18:51:47.687506Z",
+     "shell.execute_reply.started": "2021-11-09T18:51:47.675511Z"
+    },
+    "tags": []
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "True"
+      ]
+     },
+     "execution_count": 24,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "change.value"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 25,
+   "id": "bf091a24-605a-44b7-96a6-f9322953e114",
+   "metadata": {
+    "execution": {
+     "iopub.execute_input": "2021-11-09T18:51:47.691510Z",
+     "iopub.status.busy": "2021-11-09T18:51:47.690509Z",
+     "iopub.status.idle": "2021-11-09T18:51:47.705508Z",
+     "shell.execute_reply": "2021-11-09T18:51:47.703508Z",
+     "shell.execute_reply.started": "2021-11-09T18:51:47.691510Z"
+    },
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# 726        // swap gp and gp_new\n",
+    "# 727        GrB_Vector t = gp ; gp = gp_new ; gp_new = t ;\n",
+    "gp, gp_new = gp_new, gp"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f4b64496-6c8e-4a5d-abd7-11ed3527cfb8",
+   "metadata": {},
+   "source": [
+    "The algorithm repeats until a new computation is the same as the previous result.\n",
+    "\n",
+    "Here is the full python listing updated using changes from authors' paper: Yongzhe Zhang et al., J. Parallel Distributed Comput. 19 144: 14-27 (2020)):"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 26,
+   "id": "430cd67a-b0ec-4af0-9d60-e5daaa43114a",
+   "metadata": {
+    "execution": {
+     "iopub.execute_input": "2021-11-09T18:51:47.708507Z",
+     "iopub.status.busy": "2021-11-09T18:51:47.708507Z",
+     "iopub.status.idle": "2021-11-09T18:51:47.736505Z",
+     "shell.execute_reply": "2021-11-09T18:51:47.735502Z",
+     "shell.execute_reply.started": "2021-11-09T18:51:47.708507Z"
+    },
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "def fastSV(A):\n",
+    "    n = A.nrows\n",
+    "    I = np.arange(n)\n",
+    "    # The parent of each vertex is initialized to be the vertex itself:\n",
+    "    f = Vector.from_values(I.tolist(), I.tolist(), name='parents')\n",
+    "    gp = f.dup()        # grandparent of each vertex initialized to parent\n",
+    "    gp_dup = gp.dup()   # duplicate grandparents\n",
+    "    mngp = f.dup()      # minimum grandparent of each star-vertex\n",
+    "\n",
+    "    # boolean flag for each vertex\n",
+    "    mod = Vector.new(dtype=bool, size=n, name='modified?')\n",
+    "    # flag to terminate FastSV algorithm\n",
+    "    change = Scalar.from_value(True, dtype=bool, name='changed?')\n",
+    "\n",
+    "    while change.value:\n",
```

Comment:
`.value` is unnecessary here.  `while change:` works.

Reply:
Interesting! You're right, it has a __bool__() magic method.  Cool!